core/trino-main/src/main/java/io/trino/ExceededMemoryLimitException.java

```diff
@@ -40,6 +42,12 @@ public static ExceededMemoryLimitException exceededLocalUserMemoryLimit(DataSize
                 format("Query exceeded per-node memory limit of %s [%s]", maxMemory, additionalFailureInfo));
     }
 
+    public static ExceededMemoryLimitException exceededConnectorMemoryLimit(ConnectorName connectorName, DataSize maxMemory, String additionalFailureInfo)
```

Comment:
This is not a "connector memory limit". It's just the local, per-node memory limit. Rename this to `exceededLocalMemoryLimit()`.